Sources/AppcuesKit/Presentation/Traits/TraitComposer.swift

```diff
@@ -83,8 +85,20 @@ internal class TraitComposer: TraitComposing {
             stepModelsWithDecorators.append((stepModel, stepDecoratingTraits))
         }
 
+        let unwrappedPresenting = try presenting.unwrap(or: TraitError(description: "Presenting capability trait required"))
+
+        // special case - embeds - provide the trait access to the embed container view controller
+        // and vice-versa (for dismissal)
+        if let embedPresenting = unwrappedPresenting as? AppcuesEmbedTrait {
+            let embedView = resolveEmbedViewFor(experience: experience)
+            embedPresenting.embedView = embedView
+            embedView?.embedTrait = embedPresenting
+        }
```

Comment:
`try presenting.unwrap` was previously down after `wrapperCreating`, just before the `return`. Does it need to be moved up to here? (Trying to fail early perhaps?)

I'm asking because the docs list the order traits are applied (https://appcues.github.io/appcues-ios-sdk/documentation/appcueskit/traits#Trait-Application-Sequence) and I ordered them that way to try and be as easy to understand as possible (i.e. "prepare everything and then present it"). Realistically there's not many cases where the application order should matter.

Reply:
yes, it can be moved back I think. It was here only to allow the AppcuesEmbedTrait to know that the view was not found earlier, during `createWrapper` - but it would not be required to be in that order.  It can fail later in `present`.